### PR TITLE
Add explanation to customize monitoring images

### PR DESCRIPTION
**Pull request by p-se**
Fixes bsc#1177961

Signed-off-by: Patrick Seidensal <pseidensal@suse.com>

### CODE REVIEW

File: xml/admin_monitoring_alerting.xml [hunk 1]
```diff
@@ -198,6 +200,56 @@ placement:
         </step>
       </procedure>
     </sect2>
+
+    <sect2 xml:id="monitoring-custom-images">
+      <title>Deploying Using Custom or Local Images</title>
+      <para>
+        To deploy custom or local container images, the images need to be set
+        in cephadm. To do so, you will need to run the following command:
```
Comment (asettle) [inline xml/admin_monitoring_alerting.xml:208]:
```suggestion
        in &cephadm;. To do so, you will need to run the following command:

File: xml/admin_monitoring_alerting.xml [hunk 2]
```diff
@@ -198,6 +200,56 @@ placement:
...
+      <para>
+        By default, the following images are used as <literal>value</literal>:
+      </para>
+      <itemizedlist>
+        <listitem><para>registry.suse.com/caasp/v4.5/prometheus-server:2.18.0</para></listitem>
+        <listitem><para>registry.suse.com/caasp/v4.5/prometheus-node-exporter:0.18.1</para></listitem>
+        <listitem><para>registry.suse.com/caasp/v4.5/prometheus-alertmanager:0.16.2</para></listitem>
+        <listitem><para>registry.suse.com/ses/7/ceph/grafana:7.0.3</para></listitem>
+      </itemizedlist>
+      <para>Example:</para>
```
Comment (asettle) [inline xml/admin_monitoring_alerting.xml:230]:
```suggestion
      <para>For example:</para>

File: xml/admin_monitoring_alerting.xml [hunk 3]
```diff
@@ -198,6 +200,56 @@ placement:
...
+          not overwritten). The default value changes when updates become
+          available. By setting a custom image, you will not be able to update
+          the component you have set the custom image for automatically. You
+          will need to manually update the configuration (image name and tag) to
+          be able to install updates.
+        </para>
+        <para>
+          If you choose to go with the recommendations instead, you can reset
+          the custom image you have set before. After that, the default value
+          will be used again. Use <literal>ceph config rm</literal> to reset the
```
Comment (asettle) [inline xml/admin_monitoring_alerting.xml:244]:
```suggestion
          will be used again. Use the <command>ceph config rm</command> command to reset the

File: xml/admin_monitoring_alerting.xml [hunk 4]
```diff
@@ -198,6 +200,56 @@ placement:
...
+          configuration option:
+        </para>
+        <screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/&lt;option_name&gt;</screen>
+        <para>Example:</para>
```
Comment (asettle) [inline xml/admin_monitoring_alerting.xml:248]:
```suggestion
        <para>For example:</para>

File: xml/admin_monitoring_alerting.xml [hunk 5]
```diff
@@ -198,6 +200,56 @@ placement:
...
+
+    <sect2 xml:id="monitoring-custom-images">
+      <title>Deploying Using Custom or Local Images</title>
+      <para>
+        To deploy custom or local container images, the images need to be set
+        in cephadm. To do so, you will need to run the following command:
+      </para>
+      <screen>&prompt.cephuser;ceph config set mgr mgr/cephadm/&lt;option_name&gt; &lt;value&gt;</screen>
+      <para>
+        Where <literal>&lt;option_name&gt;</literal> is any of the following names:
```
Comment (asettle) [inline xml/admin_monitoring_alerting.xml:212]:
```suggestion
        Where <replaceable>option_name</replaceable> is any of the following names:

File: xml/admin_monitoring_alerting.xml [hunk 6]
```diff
@@ -198,6 +200,56 @@ placement:
...
       </procedure>
     </sect2>
+
+    <sect2 xml:id="monitoring-custom-images">
+      <title>Deploying Using Custom or Local Images</title>
+      <para>
+        To deploy custom or local container images, the images need to be set
+        in cephadm. To do so, you will need to run the following command:
+      </para>
+      <screen>&prompt.cephuser;ceph config set mgr mgr/cephadm/&lt;option_name&gt; &lt;value&gt;</screen>
```
Comment (asettle) [inline xml/admin_monitoring_alerting.xml:210]:
```suggestion
      <screen>&prompt.cephuser;ceph config set mgr mgr/cephadm/<replaceable>option_name</replaceable> <replaceable>value</replaceable></screen>

Comment (asettle) [reply xml/admin_monitoring_alerting.xml:210]: @p-se any particular reason this one didn't go in? Otherwise, LGTM

**Review by asettle — APPROVED** [review]

File: xml/admin_monitoring_alerting.xml (hunk 6 above)
Comment (p-se) [reply xml/admin_monitoring_alerting.xml:210]: Just missed it. Still checking and resolving comments. It'll be in shortly.

Comment (p-se) [reply xml/admin_monitoring_alerting.xml]: Done

File: xml/admin_monitoring_alerting.xml [hunk 7]
```diff
@@ -198,6 +200,56 @@ placement:
...
       </procedure>
     </sect2>
+
+    <sect2 xml:id="monitoring-custom-images">
+      <title>Deploying Using Custom or Local Images</title>
+      <para>
+        To deploy custom or local container images, the images need to be set
+        in &cephadm;. To do so, you will need to run the following command:
+      </para>
+      <screen>&prompt.cephuser;ceph config set mgr mgr/cephadm/<replaceable>option_name</replaceable> <replaceable>value</replaceable></screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml:210]:
```suggestion
      <screen>&prompt.cephuser;ceph config set mgr mgr/cephadm/<replaceable>OPTION_NAME</replaceable> <replaceable>VALUE</replaceable></screen>
```

File: xml/admin_monitoring_alerting.xml [hunk 8]
```diff
@@ -198,6 +200,56 @@ placement:
...
+      <para>
+        Where <replaceable>option_name</replaceable> is any of the following names:
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml:212]:
```suggestion
        Where <replaceable>OPTION_NAME</replaceable> is any of the following names:
```

File: xml/admin_monitoring_alerting.xml [hunk 9]
```diff
@@ -198,6 +200,56 @@ placement:
...
+      </para>
+
+      <itemizedlist>
+        <listitem><para>container_image_prometheus</para></listitem>
+        <listitem><para>container_image_node_exporter</para></listitem>
+        <listitem><para>container_image_alertmanager</para></listitem>
+        <listitem><para>container_image_grafana</para></listitem>
+      </itemizedlist>
+      <para>
+        By default, the following images are used as <literal>value</literal>:
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml:222]:
```suggestion
        By default, the following images are used as <replaceable>VALUE</replaceable>:
```

File: xml/admin_monitoring_alerting.xml [hunk 10]
```diff
@@ -198,6 +200,56 @@ placement:
...
+          will need to manually update the configuration (image name and tag) to
+          be able to install updates.
+        </para>
+        <para>
+          If you choose to go with the recommendations instead, you can reset
+          the custom image you have set before. After that, the default value
+          will be used again. Use <command>ceph config rm</command> to reset the
+          configuration option:
+        </para>
+        <screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/<replaceable>option_name</replaceable></screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml:247]:
```suggestion
<screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/<replaceable>option_name</replaceable></screen>
```

File: xml/admin_monitoring_alerting.xml [hunk 11]
```diff
@@ -198,6 +200,56 @@ placement:
...
+        <para>For example:</para>
+        <screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/container_image_prometheus</screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml:249]:
```suggestion
<screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/container_image_prometheus</screen>
```

File: xml/deploy_cephadm.xml [hunk 12]
```diff
@@ -702,6 +702,25 @@ o- time_server ................................................ [enabled]
...
+   <note>
+     <para>
+       For the monitoring stack, there are more container images needed. For an
+       air-gapped deployment as well as deploying from a local registry, you may
+       want to get those images at this point to prepare your local registry
+       accordingly.
+     </para>
+     <para>
+       Please note that they will not be used by &cephsalt; for the deployment.
+       Deploying monitoring services is part of the Operations and
```
Comment (tbazant) [inline xml/deploy_cephadm.xml:714]: - really the Admin guide? i thought it's https://susedoc.github.io/doc-ses/master/html/ses-deployment/deploy-cephadm.html#deploy-cephadm-day2-service-monitoring
- anyways, attach corresponding link please

**Review by tbazant — CHANGES_REQUESTED** [review]
few suggestions

File: xml/admin_monitoring_alerting.xml (hunk 10 above)
Comment (p-se) [reply xml/admin_monitoring_alerting.xml:247]: There isn't a difference in the result of the document when the indentation is not consistent. Is there another reason for suggesting to not having a consistent indentation?

Comment (tbazant) [reply xml/admin_monitoring_alerting.xml:247]: We always shift <screen>'s to the very left, our XMLformat config file is setup that way, just a professional deformaiton :-)

File: xml/admin_monitoring_alerting.xml [hunk 13]
```diff
@@ -198,6 +200,56 @@ placement:
...
+          will need to manually update the configuration (image name and tag) to
+          be able to install updates.
+        </para>
+        <para>
+          If you choose to go with the recommendations instead, you can reset
+          the custom image you have set before. After that, the default value
+          will be used again. Use <command>ceph config rm</command> to reset the
+          configuration option:
+        </para>
+<screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/<replaceable>option_name</replaceable></screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml:247]:
```suggestion
<screen>&prompt.cephuser;ceph config rm mgr mgr/cephadm/<replaceable>OPTION_NAME</replaceable></screen>
```

**Review by lproven — APPROVED** [review]

File: xml/deploy_cephadm.xml [hunk 14]
```diff
@@ -702,6 +702,25 @@ o- time_server ................................................ [enabled]
...
+       For the monitoring stack, there are more container images needed. For an
+       air-gapped deployment as well as deploying from a local registry, you may
+       want to get those images at this point to prepare your local registry
+       accordingly.
+     </para>
+     <para>
+       Please note that those container images will not be used by &cephsalt;
+       for the deployment. It is just a preparation for a later step where
+       &cephadm; will be used for the deployment or migration of monitoring
+       components.
```
Comment (p-se) [inline xml/deploy_cephadm.xml:716]: I've updated this one as discussed. You might want to review before merging it.

Comment (tbazant) [reply xml/deploy_cephadm.xml:716]: good

**Review by tbazant — APPROVED** [review]
lgtm

**Pull request merged**